### DISCUSSION TITLE
Visibility algorithms

### DISCUSSION THREAD

**Discussion by ramcdona** [elalish/manifold#601] [General]
I've been reading a bit lately about object-space visibility algorithms.  This seems to be a lesser trod area of computational geometry.

The literature says that the object space visibility problem is a 5-dimensional CSG problem in Plucker space (whatever that means).

So far, the only open source 3D object space visibility library I've found is [visilib](https://github.com/dhaumont/visilib).

Would the ideas implemented in Manifold from Julian Smith's dissertation extend to the 5-D Visibility problem?  It seems that if these ideas really generalize, then it 'ought to be' possible to develop some sort of generic N-dimensional CSG algorithm that could handle the range of problems from Clipper to Manifold to Visilib -- and maybe more.

**Comment by elalish** [comment]
I would guess so - his technique is based on building up through the dimensions, so I think it would naturally generalize, though I don't feel like figuring out all the math myself. It would certainly work in 2D as well, but it was quicker to integrate Clipper than do it myself.

**Reply by ramcdona** [reply]
That is kinda what I thought.  I certainly wasn't suggesting it as a development direction for Manifold or you.

Perhaps some future computer geometry grad student will decide to take that approach.

**Reply by elalish** [reply]
Yeah, it would certainly be cool to see someone extend it!
